server/routerlicious/packages/lambdas/src/scribe/summaryReader.ts

```diff
@@ -33,18 +33,28 @@ export class SummaryReader implements ISummaryReader {
                 const wholeFlatSummary = await this.summaryStorage.getSummary(existingRef.object.sha);
                 const normalizedSummary = convertWholeFlatSummaryToSnapshotTreeAndBlobs(wholeFlatSummary);
 
-                // Parse specific fields from the downloaded summary
+                // Obtain IDs of specific fields from the downloaded summary
                 const attributesBlobId = normalizedSummary.snapshotTree.trees[".protocol"].blobs.attributes;
+                const scribeBlobId = normalizedSummary.snapshotTree.trees[".serviceProtocol"]?.blobs?.scribe;
+                const deliBlobId = normalizedSummary.snapshotTree.trees[".serviceProtocol"]?.blobs?.deli;
+                const opsBlobId = normalizedSummary.snapshotTree.trees[".logTail"]?.blobs?.logTail;
+
+                // The initial summary uploaded by Alfred has only .protocol. In other words, .serviceProtocol
+                // and .logTail would be both missing in that scenario and we should return the default summary
+                // state if that is the case.
+                if (!scribeBlobId && !deliBlobId && !opsBlobId) {
+                    summaryReaderMetric.success(`Returning default summary when trying to read initial whole summary`);
+                    return this.getDefaultSummaryState();
+                }
```

Comment:
We are making an assumption that either all three of them are present or none. This might be true today but I am not sure whether we should rely on it. Rather than doing that, we can only return default values for missing stuff. For example, "messages" in ILatestSummaryState comes from ".logtail" blob. So if logtail is missing, we only use the default value of messages field but use the rest from deli and scribe blob.

But I am okay if we check in this one for now and open a new one for filling specific field.

Reply:
Sure. I tried to accomplish that in the latest iteration. The part I'm not sure about now is `fromSummary: true` in the `ILatestSummaryState`. Now, it might not be 100% true that the data being read is from the actual summary. Is that a concern?

Reply:
I believe fromSummary needs to be true (https://github.com/microsoft/FluidFramework/blob/main/server/routerlicious/packages/lambdas/src/scribe/lambdaFactory.ts#L125). Otherwise we will return the whole DefaultScribe object. Double-check just to be confirmed.

Reply:
Correct - and that would also cause trouble to the lambda: https://github.com/microsoft/FluidFramework/blob/73f53a3c0247c8bf2c065848f085a50df6535deb/server/routerlicious/packages/lambdas/src/scribe/lambda.ts#L119

I will leave that as is then :)